lib/promote_release.js

```diff
@@ -0,0 +1,437 @@
+      } else {
+        cli.error('Local clone is not ready');
+        throw new Error('Aborted');
+      }
+    }
+
+    await this.parseDataFromReleaseCommit();
+
+    const { version } = this;
+    cli.startSpinner('Verifying Jenkins CI status');
+    if (!jenkinsReady) {
+      cli.stopSpinner(
+        `Jenkins CI is failing for #${prid}`, cli.SPINNER_STATUS.FAILED);
+      const proceed = await cli.prompt('Do you want to proceed?');
+      if (!proceed) {
+        cli.warn(`Aborting release promotion for version ${version}`);
+        throw new Error('Aborted');
+      }
+    } else {
+      cli.stopSpinner('Jenkins CI is passing');
+    }
+
+    cli.startSpinner('Verifying GitHub CI status');
+    if (!githubCIReady) {
+      cli.stopSpinner(
+        `GitHub CI is failing for #${prid}`, cli.SPINNER_STATUS.FAILED);
+      const proceed = await cli.prompt('Do you want to proceed?');
+      if (!proceed) {
+        cli.warn(`Aborting release promotion for version ${version}`);
+        throw new Error('Aborted');
+      }
+    } else {
+      cli.stopSpinner('GitHub CI is passing');
+    }
+
+    cli.startSpinner('Verifying PR approval status');
+    if (!isApproved) {
+      cli.stopSpinner(
+        `#${prid} does not have sufficient approvals`,
+        cli.SPINNER_STATUS.FAILED);
+      const proceed = await cli.prompt('Do you want to proceed?');
+      if (!proceed) {
+        cli.warn(`Aborting release promotion for version ${version}`);
+        throw new Error('Aborted');
+      }
+    } else {
+      cli.stopSpinner(`#${prid} has necessary approvals`);
+    }
+
+    // Create and sign the release tag.
+    const shouldTagAndSignRelease = await cli.prompt(
+      'Tag and sign the release?');
+    if (!shouldTagAndSignRelease) {
+      cli.warn(`Aborting release promotion for version ${version}`);
+      throw new Error('Aborted');
+    }
+    await this.secureTagRelease();
+
+    // Set up for next release.
+    cli.startSpinner('Setting up for next release');
+    await this.setupForNextRelease();
+    cli.stopSpinner('Successfully set up for next release');
+
+    const shouldMergeProposalBranch = await cli.prompt(
+      'Merge proposal branch into staging branch?');
+    if (!shouldMergeProposalBranch) {
+      cli.warn(`Aborting release promotion for version ${version}`);
+      throw new Error('Aborted');
+    }
+
+    // Merge vX.Y.Z-proposal into vX.x.
+    cli.startSpinner('Merging proposal branch');
+    await this.mergeProposalBranch();
+    cli.stopSpinner('Merged proposal branch');
+
+    // Cherry pick release commit to master.
+    const shouldCherryPick = await cli.prompt(
+      'Cherry-pick release commit to the default branch?', { defaultAnswer: true });
+    if (!shouldCherryPick) {
+      cli.warn(`Aborting release promotion for version ${version}`);
+      throw new Error('Aborted');
+    }
+    await this.cherryPickToDefaultBranch();
```

Comment:
Shouldn't it be encapsulated in a try/catch? Also, we might want to update the session status to --continue after an error

Reply:
A failed cherry-pick does not throw, so the try/catch seems unnecessary. I agree the `--continue` would be nice, but that does feel more involved than I'd like, maybe it can be a follow up.

Reply:
Alright